htdocs/expensereport/class/paymentexpensereport.class.php

```diff
@@ -113,7 +113,7 @@ public function create($user)
 		$now = dol_now();
 
 		// Validate parameters
-		if (!$this->datepaid) {
+		if (!$this->datep) {
```

Comment:
If a make sa search on ->datepaid in dir expensereport i see found a 
$payment->datepaid       = $datepaid;
just before a 
$payment->create()
This means the property ->datepaid is set before calling the create, so if you change the test here, you will probably get an error.

There is surely a problem with datepaid / datep   (both my be used). But of you change at some place, you must also analyse who use the modified code and check if code after the change still works. Can you check this (by making an expense report and then creating a payment) to validate your change does not break something ?